src/Cardano/Wallet/Primitive/AddressDerivation.hs

```diff
@@ -509,7 +511,15 @@ keyToAddress
 keyToAddress (Key xpub) =
     Address $ CBOR.toStrictByteString $ encodeAddress xpub encodeAttributes
   where
-    encodeAttributes = CBOR.encodeMapLen 0
+    encodeAttributes = case network of
+        Testnet -> mempty
+            <> CBOR.encodeMapLen 1
+            <> CBOR.encodeWord 2
+            <> CBOR.encodeBytes (
+                  CBOR.toStrictByteString
+                $ encodeProtocolMagic (protocolMagic Testnet))
+        _ -> CBOR.encodeMapLen 0
```

Comment:
This isn't right. `Local` uses the same representation as `Testnet` !
Also, as a rule of thumb: try to _avoid_ wildcard in pattern match on union-types... This can be a great source of pain when extending types definition. A whole bunch of case won't be triggered by compiler and simply, silently swallowed.
I'll actually add that one to the coding standard as a proposal.